experimental/DrinfeldHeckeAlgebras/src/HelperFunctions.jl

```diff
@@ -0,0 +1,48 @@
+#######################################
+# Helper functions for generic Drinfeld-Hecke form generation and validation
+#
+# Cassandra Koenen, 2025
+#######################################
+
+#######################################
+# Solves the LES Mx = 0 and returns a parametrized solution over the given ring
+#######################################
+function solve_and_parametrize(M::MatElem{T}, R::Ring) where {T <: FieldElem}
+  nullity, kernel_basis = nullspace(M)
+  m = nrows(kernel_basis)
+  
+  # If nullity = 0, there is only the solution 0
+  if nullity == 0
+    return fill(R(), m)
+  end
+  
+  # For creating a parametrized solution, we work over the polynomial ring S = R[t1,...tn]
+  parameters = nullity == 1 ? ["t"] : ["t" * string(i) for i in 1:nullity]
+  S, _ = polynomial_ring(R, parameters)
```

Comment:
This creates a cached polynomial ring. Please have a look at https://docs.oscar-system.org/stable/DeveloperDocumentation/caching/#Rules-for-implementations